### PR TITLE
ICU-21450 Add builds for MacOS and for g++ with extra warnings to GHA CI.

### PR DESCRIPTION
**Pull request by gnrunge**
ICU-21450 Fix indent.

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21450
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: .github/workflows/icu_ci.yml [hunk 1]
```diff
@@ -32,6 +32,27 @@ jobs:
           ant init;
           ant check;
 
+  # Out of source build with gcc 8 and Wextra.
```
Comment (markusicu) [inline .github/workflows/icu_ci.yml:35]: Can we use a newer gcc version here? I suspect that "gcc 8" really wants to be the "latest available version of gcc".
https://gcc.gnu.org/releases.html

File: .github/workflows/icu_ci.yml [hunk 2]
```diff
@@ -32,6 +32,27 @@ jobs:
...
+  gcc-8-Wextra:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: ICU4C with gcc 8 and extra warnings.
+        run: |
```
Comment (markusicu) [inline .github/workflows/icu_ci.yml:42]: Just curious: What does the pipe symbol do?

File: .github/workflows/icu_ci.yml [hunk 3]
```diff
@@ -32,6 +32,27 @@ jobs:
...
+          mkdir build;
+          cd build;
+          ../icu4c/source/runConfigureICU Linux --disable-debug --enable-release --disable-layoutex --prefix=$PREFIX;
+          make -j2;
+          make -j2 check;
+          make install;
+          PATH=$PREFIX/bin:$PATH make -C test/hdrtst check
+        env:
+          PREFIX: /tmp/icu-prefix
```
Comment (markusicu) [inline .github/workflows/icu_ci.yml:51]: Optional: I assume that the order of "run" vs. "env" doesn't matter, but for human understanding I would prefer "env" before "run".

File: .github/workflows/icu_ci.yml [hunk 4]
```diff
@@ -32,6 +32,27 @@ jobs:
           ant init;
           ant check;
 
+  # Out of source build with gcc 8 and Wextra.
+  gcc-8-Wextra:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: ICU4C with gcc 8 and extra warnings.
```
Comment (markusicu) [inline .github/workflows/icu_ci.yml:41]: ... and hdrtst

File: .github/workflows/icu_ci.yml [hunk 5]
```diff
@@ -32,6 +32,27 @@ jobs:
           ant init;
           ant check;
 
+  # Out of source build with gcc 8 and Wextra.
+  gcc-8-Wextra:
```
Comment (markusicu) [inline .github/workflows/icu_ci.yml:36]: Could we reasonably add the Wextra and hrdtst into another existing job, maybe `gcc-8-stdlib14`?
(And maybe upgrade the gcc version there if a newer one is available for GHA?)

File: .github/workflows/icu_ci.yml (hunk 2 above)
Comment (sffc) [reply .github/workflows/icu_ci.yml:42]: It starts a multiline string in Yaml.

File: .github/workflows/icu_ci.yml (hunk 1 above)
Comment (gnrunge) [reply .github/workflows/icu_ci.yml]: Switched to version 10, in what is now gcc-10-stdlib, this particular job is removed in accordance with comment below.

File: .github/workflows/icu_ci.yml (hunk 3 above)
Comment (gnrunge) [reply .github/workflows/icu_ci.yml]: Rearranged.

File: .github/workflows/icu_ci.yml (hunk 4 above)
Comment (gnrunge) [reply .github/workflows/icu_ci.yml]: Job has been removed in accordance to comment below.

File: .github/workflows/icu_ci.yml (hunk 5 above)
Comment (gnrunge) [reply .github/workflows/icu_ci.yml]: -Wextra added to job gcc-10-stdlib14.
hdrtst is also executed by job gcc-debug-build-and-test.

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Pull request merged**